From VM Fri Jan 10 07:03:07 2003
X-VM-v5-Data: ([nil nil nil nil nil nil nil nil nil]
	["3589" "Thursday" "22" "March" "2001" "17:58:23" "-0800" "Caleb Crome" "caleb@crome.org" nil "134" "" "^From:" nil nil "3" nil nil nil nil nil nil nil nil]
	nil)
Message-ID: <20010323015823.4F04C10CB@sampson.crome.org>
MIME-Version: 1.0
Content-Type: text/plain;
	charset="iso-8859-1"
Content-Transfer-Encoding: 7bit
From: "Caleb Crome" <caleb@crome.org>
To: <caleb@crome.org>
Date: Thu, 22 Mar 2001 17:58:23 -0800
X-Priority: 3 (Normal)
X-MSMail-Priority: Normal
X-Mailer: Microsoft Outlook IMO, Build 9.0.2416 (9.0.2910.0)
Importance: Normal
X-MimeOLE: Produced By Microsoft MimeOLE V5.50.4133.2400
X-junkfilter: 20001130
X-Spammer: white: caleb@crome.org 

*** flexbackup-0.9.8/flexbackup	Tue Nov  2 19:13:41 1999
--- /usr/bin/flexbackup	Thu Mar 22 17:26:48 2001
***************
*** 1490,1497 ****

      # Check device (or dir)
      $main::ftape = 0 ;
!     if (-c $realdev) {
!
          # Check for ftape driver
          if ($realdev =~ /n?z?[qr]ft(\d+)/) {
              $main::ftape = 1;
--- 1490,1506 ----

      # Check device (or dir)
      $main::ftape = 0 ;
!     $main::remote = 0 ;
!     $main::dd = "dd";
!     if ($realdev =~ m/.*:.*/) {
! 	print "Assuming remote device $realdev\n";
! 	$main::remote = 1;
! 	$realdev =~ m/^(.*):(.*)$/;
! 	$main::device = $2;
! 	$main::remote_host = $1;
! 	$main::dd = "rsh $main::remote_host dd";
!     } elsif (-c $realdev) {
!
          # Check for ftape driver
          if ($realdev =~ /n?z?[qr]ft(\d+)/) {
              $main::ftape = 1;
***************
*** 1507,1516 ****
      } else {
          push(@main::errors,"\$device must be set to a directory or a
device");
      }
-     $main::device = $cfg::device;

      # Can we write to it?
!     if (! -w $main::device) {
          push(@main::errors,"Can't write to $main::device");
      }

--- 1516,1524 ----
      } else {
          push(@main::errors,"\$device must be set to a directory or a
device");
      }

      # Can we write to it?
!     if (!$main::remote && ! -w $main::device) {
          push(@main::errors,"Can't write to $main::device");
      }

***************
*** 1520,1526 ****
          $main::mt = 'ftmt';
      } else {
          &checkinpath('mt');
!         $main::mt = 'mt';
      }

      # Exclude regexp for find
--- 1528,1538 ----
          $main::mt = 'ftmt';
      } else {
          &checkinpath('mt');
! 	if ($main::remote) {
! 	    $main::mt = "rsh $main::remote_host mt";
! 	} else {
! 	    $main::mt = 'mt';
! 	}
      }

      # Exclude regexp for find
***************
*** 1666,1674 ****
          }
      } else {
          &checkinpath('dd');
!         $main::buffer_cmd = " | dd $main::dd_blk_flag $main::dd_pad_flag";
          $main::buffer_write_flags = "of=";
!         $main::read_cmd = "dd $main::dd_blk_flag $main::dd_pad_flag if=";
      }

      # Check for type of touch flag
--- 1678,1686 ----
          }
      } else {
          &checkinpath('dd');
!         $main::buffer_cmd = " | $main::dd $main::dd_blk_flag
$main::dd_pad_flag";
          $main::buffer_write_flags = "of=";
!         $main::read_cmd = "$main::dd $main::dd_blk_flag $main::dd_pad_flag
if=";
      }

      # Check for type of touch flag
***************
*** 2450,2456 ****

      &log("| Creating index key $key");
      $string = "printf \'$key\\nThis is a flexbackup index key\\n\' | ";
!     $string .=  "dd $main::dd_blk_flag $main::dd_pad_flag of=$dev";

      if (defined($main::opt{'n'})) {
          &log("| (debug) $string");
--- 2462,2468 ----

      &log("| Creating index key $key");
      $string = "printf \'$key\\nThis is a flexbackup index key\\n\' | ";
!     $string .=  "$main::dd $main::dd_blk_flag $main::dd_pad_flag of=$dev";

      if (defined($main::opt{'n'})) {
          &log("| (debug) $string");
***************
*** 2501,2507 ****

      } else {

!         $string = "dd $main::dd_blk_flag $main::dd_pad_flag count=1
if=$main::device";
          if (defined($main::opt{'n'})) {
              &log("| (debug) $string");
              $key = '';
--- 2513,2519 ----

      } else {

!         $string = "$main::dd $main::dd_blk_flag $main::dd_pad_flag count=1
if=$main::device";
          if (defined($main::opt{'n'})) {
              &log("| (debug) $string");
              $key = '';

